integration-cli/daemon/daemon.go

```diff
@@ -757,6 +759,16 @@ func (d *Daemon) ReloadConfig() error {
 	return nil
 }
 
+// NewClient creates new client based on daemon's socket path
+func (d *Daemon) NewClient() (*client.Client, error) {
+	var httpClient *http.Client
```

Comment:
`NewHTTPClient()` sets up the transport to be more similar to what we have in the client.

We also might add more to that method, and we shouldn't have two ways to create an httpClient in tests.